src/veh_interact.cpp

```diff
@@ -316,30 +315,19 @@ task_reason veh_interact::cant_do (char mode)
         } );
         has_tools = true; // checked later
         break;
-    case 'f': // refill mode
-        if (!ptanks.empty()) {
-            std::vector<vehicle_part *>::iterator iter;
-            for (iter = ptanks.begin(); iter != ptanks.end(); ) {
-                if ((*iter)->hp > 0 &&
-                    g->refill_vehicle_part(*veh, *iter, true)) {
-                    pass_checks = true;
-                    iter++;
-                } else {
-                    iter = ptanks.erase(iter);
-                }
-            }
-            if(pass_checks) {
-                return CAN_DO;
-            }
-        }
 
-        // if refillable parts exist but can't be refilled for some reason.
-        if (has_ptank) {
-            return CANT_REFILL;
-        }
+    case 'f': {
+        // first find all possible ammo
+        auto opts = g->u.items_with( []( const item &e ) { return e.is_ammo(); } );
```

Comment:
> What part of the ammo data is required by the tank vehicle part?

The contents must be counted by charges and therefore either `AMMO` or `COMESTIBLE`.

> YAGNI is my argument as well.

Your suggestion enables support for something that we don't make any use of either in mainline  content or indeed any mods.

> why does a liquid need ammo data just to store it in a vehicle tank?

Yes, with time we probably need to separate out `AMMO` into it's different facets but this affects pretty much the entire codebase. That wouldn't be appropriate for a bugfix.